### ISSUE TITLE
Telescope on strings

### ISSUE THREAD

**Issue by zachriggle** [pwndbg/pwndbg#368]
We should display more of this string, perhaps?

```
 X20  0x7cccbe8cdb ◂— 0x6c656e72656b5f5f ('__kernel')
...
pwndbg> ds $x20
7cccbe8cdb u'__kernel_clock_gettime'
```

**Comment by JohnPeng47** [comment]
Hi I just recently started using pwndbg and found it a great improvement over vanilla gdb for exploit development. I would like to give back to the community and thought this seemed like a good issue to start. Could you give me some more details?

**Comment by disconnect3d** [comment]
@JohnPeng47 Hey, nice! :)

So... the thing here is that `telescope` shows multiple addresess and was mainly (?) used to display stack (but we also have a command `stack`, heh) - see below - telescope used no a stack address containing `argv[0]`:
```
pwndbg> telescope 0x7fffffffda67
00:0000│   0x7fffffffda67 ◂— 0x63642f656d6f682f ('/home/dc')
01:0008│   0x7fffffffda6f ◂— '/tmp/a.out'
02:0010│   0x7fffffffda77 ◂— 0x4f435f534c007475 /* 'ut' */
03:0018│   0x7fffffffda7f ◂— 0x3d6f6e3d53524f4c ('LORS=no=')
04:0020│   0x7fffffffda87 ◂— 0x3b353b38333b3030 ('00;38;5;')
05:0028│   0x7fffffffda8f ◂— 0x303d73723a343432 ('244:rs=0')
06:0030│   0x7fffffffda97 ◂— 0x333b30303d69643a (':di=00;3')
07:0038│   0x7fffffffda9f ◂— 0x6c3a33333b353b38 ('8;5;33:l')
```

So... the question is - should we change it - and if so - how?

Maybe in such case as shown above we should display it as:
```
pwndbg> telescope 0x7fffffffda67
00:0000│   0x7fffffffda67 ◂— '/home/dc/tmp/a.out'
02:0010│   0x7fffffffda77 ◂— 0x4f435f534c007475 /* 'ut' */
```

@zachriggle @anthraxx @cebrusfs and anyone who wants to influence on this, please comment.

**Comment by JohnPeng47** [comment]
Ok also I havent really looked too much into the source code so would you be able to give me a brief overview of what is what?  I'm planning on dedicating a decent chunk of time on pwndbg so I am looking to get decent understanding of the source code

**Comment by disconnect3d** [comment]
@JohnPeng47 Added some developer notes in https://github.com/pwndbg/pwndbg/pull/405 - they can be found here: https://github.com/pwndbg/pwndbg/blob/dev/DEVELOPING.md

If you have some other questions, feel free to ask in an issue or on our freenode's irc channel (`#pwndbg`) - I might be a bit inaccessible today/tomorrow but there is also @anthraxx or @zachriggle ;).

**Comment by JohnPeng47** [comment]
Working on a small fix now

**Comment by disconnect3d** [comment]
Btw the length of display is controlled by `pwndbg/__init__.py:set print elements 15`

I think 15 is way too low, it is a bit annoying sometimes BUT we do it a bit on purpose not to overextend the screen (e.g. go to another line).

We should probably set it to something bigger like 100 or 200 and when we display something in our UI, truncate the strings on our own (if needed).

**Comment by zachriggle** [comment]
I agree that it would be nice to display strings as long as possible without wrapping, but I'm not sure how to implement it gracefully.

**Comment by disconnect3d** [comment]
Nobody really bothered about this for years. Closing.
